### PR TITLE
MAP-365 Remove legacy non-associations code

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/LegacyResource.kt [hunk 1]
```diff
@@ -31,51 +29,6 @@ import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.service.NonAssociati
 class LegacyResource(
   val nonAssociationsService: NonAssociationsService,
 ) {
-  @GetMapping("/offenders/{prisonerNumber}/non-association-details")
```
Comment (xoen) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/LegacyResource.kt:34]: I don't see any requests to this in AppInsights for the last week 🎊

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/dto/LegacyRestrictionType.kt [hunk 2]
```diff
@@ -8,20 +8,5 @@ package uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto
 enum class LegacyRestrictionType(val description: String) {
   CELL("Do Not Locate in Same Cell"),
   LAND("Do Not Locate on Same Landing"),
-  NONEX("Do Not Exercise Together"),
```
Comment (xoen) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/dto/LegacyRestrictionType.kt:11]: What happened to these restriction types?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/LegacyResource.kt (hunk 1 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/LegacyResource.kt:34]: Yes I remove the last one

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/dto/LegacyRestrictionType.kt (hunk 2 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/dto/LegacyRestrictionType.kt:11]: They don't exist in NOMIS and we never convert them back

**Review by xoen — APPROVED** [review]
LGTM

**Pull request merged**